src/contract/deploy.rs

```diff
@@ -102,6 +103,50 @@ impl<T: Transport> Builder<T> {
             waiting,
         })
     }
+
+    /// Execute deployment passing code and contructor parameters.
+    pub fn execute<P, V>(self, code: V, params: P, from: Address) -> Result<PendingContract<T>, ethabi::Error>
+    where
+        P: Tokenize,
+        V: AsRef<str>,
+        T: Transport,
+    {
+        let f = |tx, eth: T, poll_interval, confirmations| {
+            confirm::send_transaction_with_confirmation(eth, tx, poll_interval, confirmations)
+        };
+        self.execute_core(code, params, from, f)
+    }
+
+    /// Execute deployment passing code and contructor parameters.
+    ///
+    /// Unlike the above `execute`, this method uses
+    /// `sign_raw_transaction_with_confirmation` instead of
+    /// `sign_transaction_with_confirmation`, which requires the account from
+    /// which the transaction is sent to be unlocked.
+    pub fn execute_no_unlock<P, V>(
+        self,
+        code: V,
+        params: P,
+        from: Address,
+        password: &str,
+        web3: crate::Web3<T>,
+    ) -> Result<PendingContract<T>, ethabi::Error>
+    where
+        P: Tokenize,
+        V: AsRef<str>,
+        T: Transport,
+    {
+        let f = |tx, eth: T, poll_interval, confirmations| {
+            let raw_tx = web3
+                .personal()
+                .sign_transaction(tx, password)
+                .wait()
+                .expect("Transaction signing failed");
```

Comment:
We definitely should not panic if we can't sign the transaction - it should just be a regular `Err` that can be later handled by the caller.

Reply:
You are right.